sync2jira/upstream_issue.py

```diff
@@ -188,13 +188,14 @@ def github_issues(upstream, config):
         .get('github', {})\
         .get(upstream, {})
 
-    if 'labels' in _filter:
-        # We have to flatten the labels list to a comma-separated string
-        _filter['labels'] = ','.join(_filter['labels'])
-
     url = 'https://api.github.com/repos/%s/issues' % upstream
     if _filter:
-        url += '?' + urlencode(_filter)
+        # We have to flatten the labels list to a comma-separated string
+        url_filter = {
+            key: ','.join(expected) if key == 'labels' else expected
+            for key, expected in _filter.items()
+        }
```

Comment:
I don't think this is enough to "save" us.  The problem is that `join()` takes an _iterable_, and you're hoping/expecting that `expected` is a `list`; however, `str`'s are _also_ iterable, and so if `expected` is a(n already flattened) string, then `join()` won't object -- it'll just insert a comma between each character:
```
>>> ','.join("string")
's,t,r,i,n,g'
```

I think the question that we want to be asking is, is `_filter['labels']` a `list` (or maybe a `tuple`, but, definitely, is it _not_ a `str`)?  If so, then flatten it; otherwise, take it as it is.

That said, the fact that `_filter` could refer to part of `config` does suggest that we should be making a copy instead of mutating `_filter` itself (although, if the conditionalization were correct, mutating `_filter` might save us time on subsequent issues...).

I might be inclined to try code like this:
```suggestion
       if isinstance(_filter.get('labels'), list):
            _filter['labels'] = ','.join(_filter['labels'])
```
(Or, you can do the equivalent on a `deepcopy()` of `_filter`, if you're not comfortable mutating `config`.)

Reply:
> think the question that we want to be asking is, is _filter['labels'] a list (or maybe a tuple, but, definitely, is it not a str)? If so, then flatten it; otherwise, take it as it is.

I had proposed checking if the type of `_filter['labels']` was string, and just taking as-is - but it was decided to just assume it's always a list. Since the check is easy and would also avoid repeated processing I agree we should do that.

> I might be inclined to try code like this:

That's fine as well, I wanted to be extra safe thus the new object - but this should work as well. I've updated the change with your suggestion, thanks.

Reply:
> it was decided to just assume it's always a list.

Clearly that was a mistake!  😀  (We all failed to notice that we were mutating the configuration object, such that we could end up receiving a string quite easily!)

> I wanted to be extra safe thus the new object

That's generally a good and appropriate thing to do.  It's just that, in this particular case, we can save ourselves work by mutating the old object, and so it's worth considering doing so...if we can do it safely/correctly.  😉

Reply:
@webbnh I very quickly wrote up a simple unit test, let me know if that's ok too - thanks again!